precepte-core-cats/src/main/scala/package.scala

```diff
@@ -39,9 +40,15 @@ package object corecats extends SubMeta {
       override def flatMap[A, B](m: Precepte[Ta, ManagedState, UnmanagedState, F, A])(f: A => Precepte[Ta, ManagedState, UnmanagedState, F, B]): Precepte[Ta, ManagedState, UnmanagedState, F, B] =
         m.flatMap(f)
 
-      override def ap[A, B](pa: Precepte[Ta, ManagedState, UnmanagedState, F, A])(pab: Precepte[Ta, ManagedState, UnmanagedState, F, A => B]): Precepte[Ta, ManagedState, UnmanagedState, F, B] = {
+      override def ap[A, B](pab: Precepte[Ta, ManagedState, UnmanagedState, F, A => B])(pa: Precepte[Ta, ManagedState, UnmanagedState, F, A]): Precepte[Ta, ManagedState, UnmanagedState, F, B] = {
         Apply(pa, pab)
       }
+      override def tailRecM[A, B](a: A)(f: A => Precepte[Ta, ManagedState, UnmanagedState, F, Either[A,B]]): Precepte[Ta, ManagedState, UnmanagedState, F, B] = {
+        f(a).flatMap {
+          case Left(result) => tailRecM(result)(f)
+          case Right(result) => pure(result)
+        }
+      }
```

Comment:
Is this guaranteed to be stack safe?

Reply:
It's not, cf : https://github.com/typelevel/cats/blob/master/docs/src/main/tut/faq.md#tailrecm

Reply:
Je viens de le remplacer par `fastFlatMap` même si il n'est pas tail rec.